codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -38,24 +38,30 @@
  */
 public class DefaultHttp2RemoteFlowController implements Http2RemoteFlowController {
     private static final int MIN_WRITABLE_CHUNK = 32 * 1024;
-    private final Http2StreamVisitor WRITE_ALLOCATED_BYTES = new Http2StreamVisitor() {
+    private final StreamWriteDistributor.Writer WRITER = new StreamWriteDistributor.Writer() {
         @Override
-        public boolean visit(Http2Stream stream) {
-            int written = state(stream).writeAllocatedBytes();
+        public void write(Http2Stream stream, int numBytes) {
+            int written = state(stream).writeAllocatedBytes(numBytes);
             if (written != -1 && listener != null) {
                 listener.streamWritten(stream, written);
             }
-            return true;
         }
     };
     private final Http2Connection connection;
     private final Http2Connection.PropertyKey stateKey;
+    private final StreamWriteDistributor streamWriteDistributor;
     private int initialWindowSize = DEFAULT_WINDOW_SIZE;
     private ChannelHandlerContext ctx;
     private Listener listener;
 
     public DefaultHttp2RemoteFlowController(Http2Connection connection) {
+        this(connection, new PriorityStreamWriteDistributor(connection));
+    }
+
+    public DefaultHttp2RemoteFlowController(Http2Connection connection,
+                                            StreamWriteDistributor streamWriteDistributor) {
```

Comment:
`StreamWriteDistributor` -> `StreamByteDistributor`? We are distributing `bytes` not `writes` ?